parsing/src/recipes.ts

```diff
@@ -142,32 +139,138 @@ function getRecipes(
-        id: "PlutoniumWaste",
-        displayName: "Plutonium Waste",
-        ingredients: [{ part: 'PlutoniumFuelRod', amount: 1, perMin: 0.1 }, { part: 'Water', amount: 2400, perMin: 240 }],
-        products: [{ part: "PlutoniumWaste", amount: 1, perMin: 1 }],
-        building: { name: "nuclearpowerplant", power: 0 },
-        isAlternate: false,
-        isFicsmas: false
-    });
+    // // Manually add Nuclear waste recipes
+    // recipes.push({
+    //     id: "NuclearWaste",
+    //     displayName: "Uranium Waste",
+    //     ingredients: [{ part: 'NuclearFuelRod', amount: 1, perMin: 0.2 }, { part: 'Water', amount: 1200, perMin: 240 }],
+    //     products: [{ part: "NuclearWaste", amount: 1, perMin: 10 }],
+    //     building: { name: "nuclearpowerplant", power: 2500 },
+    //     isAlternate: false,
+    //     isFicsmas: false,
+    //     isPowerGenerator: true
+    // });
+    // recipes.push({
+    //     id: "PlutoniumWaste",
+    //     displayName: "Plutonium Waste",
+    //     ingredients: [{ part: 'PlutoniumFuelRod', amount: 1, perMin: 0.1 }, { part: 'Water', amount: 2400, perMin: 240 }],
+    //     products: [{ part: "PlutoniumWaste", amount: 1, perMin: 1 }],
+    //     building: { name: "nuclearpowerplant", power: 2500 },
+    //     isAlternate: false,
+    //     isFicsmas: false,
+    //     isPowerGenerator: true
+    // });
+
+    return recipes.sort((a, b) => a.displayName.localeCompare(b.displayName));
+}
+
+function getPowerGeneratingRecipes(
+    data: any[],
+    parts: PartDataInterface
+): Recipe[] {
+
+    const recipes: any[] = [];
+
+    data
+        .filter((entry: any) => entry.Classes)
+        .flatMap((entry: any) => entry.Classes)
+        .filter((recipe: any) => {
+            
+            // Filter out recipes that don't have a fuel component
+            if (!recipe.mFuel)  { 
+                return false; 
+            } else {
+                return true;
+            }
+
+        })
+        .forEach((recipe: any) => {
+             
+            let building : Building = {
+                name: recipe.mDisplayName.replace(/ /g, ''), // Use the first valid building, or empty string if none
+                power: Math.round(recipe.mPowerProduction), // generated power - can be rounded to the nearest whole number (all energy numbers are whole numbers) 
+            };   
+            let supplementalRatio: number = Number(recipe.mSupplementalToPowerRatio);
+            // 1. Generator MW generated. This is an hourly value.
+            // 2. Divide by 60, to get the minute value
+            // 3. Now calculate the MJ, using the MJ->MW constant (1/3600), (https://en.wikipedia.org/wiki/Joule#Conversions) 
+            // 4. Now divide this number by the part energy to calculate how many pieces per min
+            const powerMJ: number = (recipe.mPowerProduction / 60) / (1/3600)
+
+            const fuels: Fuel[] = recipe.mFuel       
+            fuels.forEach((fuel: any) => {
```

Comment:
```suggestion
            fuels.forEach((fuel: Fuel) => {
```
Strictly speaking this typescast should not be needed